File: domain/src/main/java/com/mrp/sml/domain/repository/TransferProgress.java
package com.mrp.sml.domain.repository;

public class TransferProgress {
    private final long transferredBytes;
    private final long totalBytes;
    private final double speedBytesPerSecond;
    private final float progressPercent;

    public TransferProgress(long transferredBytes, long totalBytes, double speedBytesPerSecond, float progressPercent) {
        this.transferredBytes = transferredBytes;
        this.totalBytes = totalBytes;
        this.speedBytesPerSecond = speedBytesPerSecond;
        this.progressPercent = progressPercent;
    }

    public long getTransferredBytes() { return transferredBytes; }
    public long getTotalBytes() { return totalBytes; }
    public double getSpeedBytesPerSecond() { return speedBytesPerSecond; }
    public float getProgressPercent() { return progressPercent; }
}

File: app/src/main/java/com/mrp/sml/core/models/TransferProgress.kt
package com.mrp.sml.core.models

data class TransferProgress(
    val transferredBytes: Long = 0L,
    val totalBytes: Long = 0L,
    val speedBytesPerSecond: Double = 0.0,
    val progressPercent: Float = 0f,
    val currentFileName: String = "",
    val currentFileIndex: Int = 0,
    val totalFiles: Int = 0,
    val etaSeconds: Long = 0L
) {
    val isIndeterminate: Boolean get() = totalBytes <= 0L
    val progressFraction: Float get() = if (totalBytes > 0L) (transferredBytes.toFloat() / totalBytes).coerceIn(0f, 1f) else 0f
}
